packages/fastify-siwe/src/api.test.ts

```diff
@@ -1,13 +1,28 @@
 import { expect } from 'chai'
+import createFastify, { FastifyInstance } from 'fastify'
 import { MockProvider } from 'ethereum-waffle'
-import 'jsdom-global/register'
 import { SiweMessage } from 'siwe'
-import { mock } from './mockApi'
+import { siwePlugin } from './plugin'
+import { InMemoryStore } from './InMemoryStore'
+import cookie from '@fastify/cookie'
+import { registerSiweRoutes } from './registerSiweRoutes'
+import { Wallet } from 'ethers'
 
-describe('signInWithEthereum', () => {
-  const app = mock()
-  const provider = new MockProvider({ ganacheOptions: { chain: { chainId: 1 } } as any })
-  const signer = provider.getWallets()[0]
+describe('Fastify with SIWE API', () => {
+  let app: FastifyInstance
+  let signer: Wallet
+
+  before(async () => {
+    const provider = new MockProvider({ ganacheOptions: { chain: { chainId: 1 } } as any })
+    signer = provider.getWallets()[0]
+    app = createFastify()
+
+    const store = new InMemoryStore()
+
+    app.register(cookie)
+    app.register(siwePlugin({ store }))
+    registerSiweRoutes(app, { store })
+  })
 
   it('returns correct nonce', async () => {
```

Comment:
Add a test where you start 2 apis, generate a nonce on each at the same time, check that they are different - to prevent nonce-replay attack.